### PR TITLE
Catch http lib related exceptions as HpOneViewExceptions during login

### LINKED ISSUE #309: HPOneViewException not raised when connection with paused VM fails
Hi guys,

When I try to create the OneView client but my VM is paused, an **OSError** exception is raised instead of a HPOneViewException. The HPOneViewException should not cover this type of exception in this scenario? For treat this exception is only possible if I catch an **OSError**, but is seems odd for me since this also could be used as a HPOneViewException maybe.

**Code example:**

```python
#VM paused
config = {
	"ip": "x.x.x.x",
	"credentials": {
		"userName": "user_name",
		"password": "pass"
	}
}

oneview_client = OneViewClient(config)
```

**Result:**

```shell
Traceback (most recent call last):
  File "one_view.py", line 89, in <module>
    one_view = OneView(config)
  File "one_view.py", line 12, in __init__
    self.oneview_client = OneViewClient(config)
  File "/usr/local/lib/python3.5/dist-packages/hpOneView/oneview_client.py", line 118, in __init__
    self.__connection.login(config["credentials"])
  File "/usr/local/lib/python3.5/dist-packages/hpOneView/connection.py", line 483, in login
    self.validateVersion()
  File "/usr/local/lib/python3.5/dist-packages/hpOneView/connection.py", line 79, in validateVersion
    version = self.get(uri['version'])
  File "/usr/local/lib/python3.5/dist-packages/hpOneView/connection.py", line 324, in get
    resp, body = self.do_http('GET', uri, '')
  File "/usr/local/lib/python3.5/dist-packages/hpOneView/connection.py", line 126, in do_http
    conn.request(method, path, body, http_headers)
  File "/usr/lib/python3.5/http/client.py", line 1106, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib/python3.5/http/client.py", line 1151, in _send_request
    self.endheaders(body)
  File "/usr/lib/python3.5/http/client.py", line 1102, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python3.5/http/client.py", line 934, in _send_output
    self.send(msg)
  File "/usr/lib/python3.5/http/client.py", line 877, in send
    self.connect()
  File "/usr/lib/python3.5/http/client.py", line 1252, in connect
    super().connect()
  File "/usr/lib/python3.5/http/client.py", line 849, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/usr/lib/python3.5/socket.py", line 711, in create_connection
    raise err
  File "/usr/lib/python3.5/socket.py", line 702, in create_connection
    sock.connect(sa)
OSError: [Errno 113] No route to host
```

### PR DESCRIPTION
**Pull request by fgbulsoni**
### Description
Added try/except block to validateVersion in order to catch host unreachable and timeout exceptions during login attempts to appliance.

### Issues Resolved
Fixes #309 

### Check List
- [X] All tests pass for Python 2.7+ & 3.4+(`$ tox`).
- [X] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: hpOneView/connection.py [hunk 1]
```diff
@@ -479,8 +479,11 @@ def change_initial_password(self, newPassword):
     # Login/Logout to/from appliance
     ###########################################################################
     def login(self, cred, verbose=False):
-        if self._validateVersion is False:
-            self.validateVersion()
+        try:
+            if self._validateVersion is False:
+                self.validateVersion()
+        except Exception as e:
+            raise HPOneViewException(str(e))
```
Comment (tmiotto) [inline hpOneView/connection.py]: Why do we need to cover every exception?

An InterruptedException, for instance, should not be re-raised as a HPOneViewException, these are not SDK errors

Comment (fgbulsoni) [reply hpOneView/connection.py]: Sure, what about the `no route to host` or `timeout` exceptions? Should we specifically seek those out?

Comment (tmiotto) [reply hpOneView/connection.py]: Maybe you should try:
```python
raise HPOneViewException(str(e)) from e
```
But I don't know if future will handle it well.

Comment (fgbulsoni) [reply hpOneView/connection.py]: I've just tried the `raise_from` from future, but it does not appear to handle it as it should:

On python2:
```
vagrant@ecosystem-centos7:~/dev/python-hpOneView/examples  (enhancement/catch-http-lib-exceptions-as-hponeviewexceptions)
$ python fc_networks.py
Traceback (most recent call last):
  File "fc_networks.py", line 54, in <module>
    oneview_client = OneViewClient(config)
  File "/home/vagrant/dev/python-hpOneView/hpOneView/oneview_client.py", line 119, in __init__
    self.__connection.login(config["credentials"])
  File "/home/vagrant/dev/python-hpOneView/hpOneView/connection.py", line 491, in login
    raise_from(HPOneViewException('Failure during login attempt.'), e)
  File "/usr/local/lib/python2.7/site-packages/future/utils/__init__.py", line 454, in raise_from
    raise e
HPOneViewException: Failure during login attempt.
vagrant@ecosystem-centos7:~/dev/python-hpOneView/examples  (enhancement/catch-http-lib-exceptions-as-hponeviewexceptions)

```
On Python3:
```
$ python3 fc_networks.py
Traceback (most recent call last):
  File "/home/vagrant/dev/python-hpOneView/hpOneView/connection.py", line 485, in login
    self.validateVersion()
  File "/home/vagrant/dev/python-hpOneView/hpOneView/connection.py", line 80, in validateVersion
    version = self.get(uri['version'])
  File "/home/vagrant/dev/python-hpOneView/hpOneView/connection.py", line 325, in get
    resp, body = self.do_http('GET', uri, '')
  File "/home/vagrant/dev/python-hpOneView/hpOneView/connection.py", line 127, in do_http
    conn.request(method, path, body, http_headers)
  File "/usr/local/lib/python3.5/http/client.py", line 1107, in request
    self._send_request(method, url, body, headers)
  File "/usr/local/lib/python3.5/http/client.py", line 1152, in _send_request
    self.endheaders(body)
  File "/usr/local/lib/python3.5/http/client.py", line 1103, in endheaders
    self._send_output(message_body)
  File "/usr/local/lib/python3.5/http/client.py", line 934, in _send_output
    self.send(msg)
  File "/usr/local/lib/python3.5/http/client.py", line 877, in send
    self.connect()
  File "/usr/local/lib/python3.5/http/client.py", line 1253, in connect
    super().connect()
  File "/usr/local/lib/python3.5/http/client.py", line 849, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/usr/local/lib/python3.5/socket.py", line 712, in create_connection
    raise err
  File "/usr/local/lib/python3.5/socket.py", line 703, in create_connection
    sock.connect(sa)
TimeoutError: [Errno 110] Connection timed out

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "fc_networks.py", line 54, in <module>
    oneview_client = OneViewClient(config)
  File "/home/vagrant/dev/python-hpOneView/hpOneView/oneview_client.py", line 119, in __init__
    self.__connection.login(config["credentials"])
  File "/home/vagrant/dev/python-hpOneView/hpOneView/connection.py", line 487, in login
    raise_from(HPOneViewException('Failure during login attempt.'), e)
  File "/usr/local/lib/python3.5/site-packages/future/utils/__init__.py", line 398, in raise_from
    exec(execstr, myglobals, mylocals)
  File "<string>", line 1, in <module>
hpOneView.exceptions.HPOneViewException: Failure during login attempt.
```

Python 2 ends up having a lot less understandable trace and message in this case

Comment (fgbulsoni) [reply hpOneView/connection.py]: So, after trying both the `reraise` method from `six` and the `raise_from` method from `future.utils`, both which are basically supposed to achieve [PEP 3134](https://www.python.org/dev/peps/pep-3134/) on Py2/Py3, I'm still seeing the same thing happening.

Py2 exception will remain not chained while Py3 has the chaining.

Do we want to keep this behavior @tmiotto ? Thoughts?

Another possibility: Specifically target `OSerror` as a `no route to host` and `TimeOut` Exception types, as those have been proven to be hit, and leave the rest to chance.

That way we do get better error messages, but we also do get more lines of code and still allow whatever other exception to pass through.

Both feel like tradeoffs to me, ideally the `from_raise` should've worked 😞

**Pull request merged**